flatpak_builder_lint/checks/metainfo.py

```diff
@@ -67,6 +67,12 @@ def _validate(self, path: str, appid: str) -> None:
         ):
             if not os.path.exists(appinfo_icon_path):
                 self.errors.add("appstream-missing-icon-file")
+            if not appstream.has_icon_key(appstream_path):
+                self.errors.add("appstream-missing-icon-key")
+            if appstream.is_icon_remote(appstream_path):
+                self.errors.add("appstream-remote-icon-key")
```

Comment:
Somewhat confusingly, there are apps mixing local and remote icons. We should not fail if both are defined.

Reply:
Just to clarify, you mean `type=local`?

Reply:
With the check modified this doesn't matter, it'll fail only when the remote icon entry doesn't start with Flathub url.

https://github.com/flathub-infra/flatpak-builder-lint/pull/277#discussion_r1490270136